Sample/Samples/UserUpdateViewController.swift

```diff
@@ -71,7 +71,13 @@ class UserUpdateViewController: UITableViewController {
             title: cell?.textLabel?.text ?? "",
             initialValue: cell?.detailTextLabel?.text ?? ""
         )
-        inputVC.onChange = { [unowned self] newValue in
+        inputVC.onChange = { [weak self] newValue in
+
+            guard let self = self else {
+                log.warning("Callback called while self is nil")
+                return nil
+            }
```

Comment:
I wouldn't even log. @kojiba @evsaev What is the value of logging here? There is none. I would remove all logs from `guard let self = self ...`.  Because in this happens, it most likely because the original `self` doesn't exist anymore, because it was deallocated, and so it doesn't matter anymore that the closure is completed.

Reply:
in some cases we must have `self` alive to do some job but strong ref is a no-go because of retain cycle. In this case, we don't want silently fail but crash/log to know there's an issue and we have wrong understanding of object lifetime

Reply:
Illiya has updated cases were `unowned` was used -> we were expecting `self` to be alive for some job. If it's not, there's an issue

Reply:
We will never know the objects lifetime because we don't control them fully, since an app can use our SDK in a way we don't expect. Either way, if there is one job that we need to make sure it is done, let's log on those ones **only** and not blindly log everywhere, otherwise, the logs will be a mess 🙂

Reply:
yes, agree that in couple of cases we assume that it should be alive, so better to keep logs